llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -836,10 +839,26 @@ class BranchInst : public Instruction {
 #endif
 };
 
-class LoadInst final : public Instruction {
+/// An abstract class, parent of unary instructions.
+class UnaryInstruction : public Instruction {
+protected:
+  UnaryInstruction(ClassID ID, Opcode Opc, llvm::Instruction *LLVMI,
+                   Context &Ctx)
+      : Instruction(ID, Opc, LLVMI, Ctx) {}
+
+public:
+  static bool classof(const Instruction *I) {
+    return I->getOpcode() == Instruction::Opcode::Load || isa<CastInst>(I);
+  }
+  static bool classof(const Value *V) {
+    return isa<Instruction>(V) && classof(cast<Instruction>(V));
```

Comment:
isa + cast again?
```
if (Instruction *I = dyn_cast<Instruction>(V))
  return classes(I);
return false;
```

Reply:
Isn't this a bit too verbose? Three lines instead of one. And btw this is exactly how it's implemented in InstrTypes.h.

Reply:
We had this discussion before. The manual asks for dyn_cast.

Reply:
I think you may be misinterpreting the manual. This pattern is *very* common throughout LLVM. It can't be that no one noticed that all this code does not conform with the coding style. InstrTypes.h in particular includes this exact pattern 119 times!

Reply:
Widely use is a bad argument.
"Note that you should not use an isa<> test followed by a cast<>, for that use the dyn_cast<> operator."

Reply:
although I like the less redundant dyn_cast version, there is merit to keeping code short. I would bring this up on discourse if you feel strongly about this

Reply:
> Widely use is a bad argument.

True, but if it's widely used then it probably means that either:
(i) we are not conflicting with what the manual says and we are misinterpreting it, or 
(ii) using `isa<> + cast<>` in this particular case is preferable despite what the manual says, because it is much less verbose.

In any case, I don't think we should be spending too much time arguing about this, it's just a style issue and both `isa<> + cast<>` and `dyn_cast<>` versions exist in the LLVM codebase.

Reply:
Probably, it was a common pattern to use isa + cast. Then the manual was updated: please use dyn_cast, but not all occurrences of isa +cast were converted to dyn_cast. That is why you find this pattern in many places.

I contribute to:
https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/CodeGen/GlobalISel/GenericMachineInstrs.h
We try to model LLVM-IR, but we are forced to deviations. There is also some freedom, i.e., ExtOrTrunc.
My point is : we use switches.